src/main/java/org/apache/bcel/verifier/statics/Pass2Verifier.java

```diff
@@ -162,7 +162,7 @@ public VerificationResult do_verify() {
                 every_class_has_an_accessible_superclass();
                 final_methods_are_not_overridden();
             }
-            catch (final ClassConstraintException cce) {
+            catch (final ClassConstraintException | ClassCastException cce) {
```

Comment:
Where can a CCE happen? It's odd to catch, could the CCE be checked explicitly where it actually happen with an `instanceof` test?

Reply:
I've not debugged this but this feels like the wrong approach: The CCE can be detected at the call site it is thrown and a check can be performed before that point and a properly typed exception thrown at that location. Transforming the exception here feels like an after-the-fact workaround/hack.

Reply:
Well the simplest fix would be to change the throw in CheckIndex at line 371 to ClassConstraintException.  Or change to ClassFormatException and change the catch to match.